chaoslib/litmus/spring-boot-chaos/lib/spring-boot-chaos.go

```diff
@@ -0,0 +1,329 @@
+				return err
+			}
+
+			if err := enableChaosMonkey(experimentsDetails.ChaosMonkeyPort, experimentsDetails.ChaosMonkeyPath, pod); err != nil {
+				log.Errorf("[Chaos]: Failed to enable chaos, err: %v ", err)
+				return err
+			}
+			common.SetTargets(pod.Name, "injected", "pod", chaosDetails)
+
+			log.Infof("[Chaos]:Waiting for: %vs", experimentsDetails.ChaosDuration)
+
+			endTime = time.After(timeDelay)
+		loop:
+			for {
+				select {
+				case <-signChan:
+					log.Info("[Chaos]: Revert Started")
+					if err := disableChaosMonkey(experimentsDetails.ChaosMonkeyPort, experimentsDetails.ChaosMonkeyPath, pod); err != nil {
+						log.Errorf("Error in disable chaos monkey, err: %v", err)
+					}
+					common.SetTargets(pod.Name, "reverted", "pod", chaosDetails)
+					os.Exit(1)
+				case <-endTime:
+					log.Infof("[Chaos]: Time is up for experiment: %v", experimentsDetails.ExperimentName)
+					endTime = nil
+					break loop
+				}
+			}
+			if err := disableChaosMonkey(experimentsDetails.ChaosMonkeyPort, experimentsDetails.ChaosMonkeyPath, pod); err != nil {
+				log.Errorf("Error in disable chaos monkey, err: %v", err)
+			}
+		}
+	}
+	return nil
+
+}
+
+// injectChaosInParallelMode delete the target application pods in parallel mode (all at once)
+func injectChaosInParallelMode(experimentsDetails *experimentTypes.ExperimentDetails, clients clients.ClientSets, chaosDetails *types.ChaosDetails, eventsDetails *types.EventDetails, resultDetails *types.ResultDetails) error {
+
+	// run the probes during chaos
+	if len(resultDetails.ProbeDetails) != 0 {
+		if err := probe.RunProbes(chaosDetails, clients, resultDetails, "DuringChaos", eventsDetails); err != nil {
+			return err
+		}
+	}
+
+	// signChan channel is used to transmit signal notifications.
+	signChan := make(chan os.Signal, 1)
+	// Catch and relay certain signal(s) to signChan channel.
+	signal.Notify(signChan, os.Interrupt, syscall.SIGTERM)
+
+	var endTime <-chan time.Time
+	timeDelay := time.Duration(experimentsDetails.ChaosDuration) * time.Second
+
+	select {
+	case <-signChan:
+		// stopping the chaos execution, if abort signal received
+		time.Sleep(10 * time.Second)
+		os.Exit(0)
+	default:
+		for _, pod := range experimentsDetails.TargetPodList.Items {
+			if experimentsDetails.EngineName != "" {
+				msg := "Injecting " + experimentsDetails.ExperimentName + " chaos on " + pod.Name + " pod"
+				types.SetEngineEventAttributes(eventsDetails, types.ChaosInject, msg, "Normal", chaosDetails)
+				_ = events.GenerateEvents(eventsDetails, clients, chaosDetails, "ChaosEngine")
+			}
+
+			log.InfoWithValues("[Chaos]: The Target application details", logrus.Fields{
+				"Target Pod": pod.Name,
+			})
+
+			if err := setChaosMonkeyWatchers(experimentsDetails.ChaosMonkeyPort, experimentsDetails.ChaosMonkeyPath, experimentsDetails.ChaosMonkeyWatchers, pod); err != nil {
+				log.Errorf("[Chaos]: Failed to set watchers, err: %v ", err)
+				return err
```

Comment:
```suggestion
				return errors.Errorf("[Chaos]: Failed to set watchers, err: %v ", 
```
The error will cascade from bottom to top, so it is better to return the error rather than logging and returning separately. This is used throughout litmus-go, so we can follow this here and at other places.